### ISSUE TITLE
How to pass a part of state as a prop and mutate the part in a child component?

### ISSUE THREAD

**Issue by toiroakr** [pmndrs/eslint-plugin-valtio#41]
I want to update a part of global state in a child component.
Then, eslint-plugin-valtio warn based on valtio/state-snapshot-rule.

How to implement in correct way?

sample:
```tsx
import { FunctionComponent } from "react";
import { proxy, useSnapshot } from "valtio";

const state = proxy({
  outer: {
    inner: {
      count: 0,
    },
  },
});

export const Parent: FunctionComponent = () => {
  const snapshot = useSnapshot(state);

  return (
    <>
      {snapshot.outer.inner.count}
      {/* Using proxies in the render phase would cause unexpected problems.eslint(valtio/state-snapshot-rule) */}
      <Child prop={state.outer.inner}></Child>
    </>
  );
};

const Child: FunctionComponent<{ prop: { count: number } }> = ({ prop }) => {
  const innerSnapshot = useSnapshot(prop);
  const handleClick = () => prop.count ++;

  return <button onClick={handleClick}>{innerSnapshot.count}</button>;
};
```
with
```
valtio: 1.9.0
eslint-plugin-valtio: 0.6.1
```

**Comment by dai-shi** [comment]
Thanks for reporting! This sounds like a tough one.

Passing a proxy state as a prop to child components is valid and actually a recommended pattern. So, I think it's a false positive.

Not sure how easy to solve it. I think we need to give up some invalid cases. @barelyhuman what do you think?

**Comment by toiroakr** [comment]
Thank you for your quick reply.

> Passing a proxy state as a prop to child components is valid and actually a recommended pattern. So, I think it's a false positive.

I'm relieved.

**Comment by barelyhuman** [comment]
> Not sure how easy to solve it. I think we need to give up some invalid cases. @barelyhuman what do you think?

Oh this would be hard to handle, I'll give it a try but if not we'll have to disable the proxy in render checks
